Userland/Libraries/LibThreading/ConditionVariable.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2021, kleines Filmröllchen <malu.bertsch@gmail.com>.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Function.h>
+#include <LibThreading/Mutex.h>
+#include <pthread.h>
+#include <sys/cdefs.h>
+#include <sys/types.h>
+
+namespace Threading {
+
+// A signaling condition variable that wraps over the pthread_cond_* APIs.
+class ConditionVariable {
+    friend class Mutex;
+
+public:
+    ConditionVariable(Mutex& to_wait_on)
+        : m_to_wait_on(to_wait_on)
+    {
+        pthread_cond_init(&m_condition, nullptr);
+    }
+
+    ALWAYS_INLINE ~ConditionVariable()
+    {
+        pthread_cond_destroy(&m_condition);
+    }
+
+    // As with pthread APIs, the mutex must be locked or undefined behavior ensues.
+    ALWAYS_INLINE void wait()
+    {
+        pthread_cond_wait(&m_condition, &m_to_wait_on.m_mutex);
```

Comment:
Technically I guess all these functions can return an error, let's `VERIFY()` that they don't.